### PR TITLE
RCF-33, RCF-99, 101, 102, 103, 104, 115

### PR DESCRIPTION
**Pull request by Piyush7034**
### RCF-33 Created a credentials page for displaying the machine credentials on long pressing of the logo.

RCF-99: Created AboutActivityService class which contains methods for generating the machine details using services defined in java packages and then storing the details in JSON and returning the JSON response through MethodChannel.
RCF-101: Added a Copy Button on credentials page which will copy the text containing machine details.
RCF-102: Unit tests written for widget testing functionalities.
RCF-103: Added a Download Json button on credentials page which will save the file in downloads of mobile.
RCF-104: Updated .md file with information on using java packages, method channels and machine details fetching.
RCF-115: Initial flutter project bootstrapping.

### CODE REVIEW

**Comment by Piyush7034** [comment]
New Login UI Integrated. Added Copy Text and download json for machine details. Added responsive views for tablet and mobile view. Added all the java packages instead of aar files. Created utils folder for defining app styles and colors.

**Comment by Piyush7034** [comment]
RCF-33 Integrated Login Flow with new login ui and added necessary validations such as Username and password is required, Length Exceed Error, Incorrect Username and password error. The ui is made responsive for both mobile and tablet.

**Comment by Piyush7034** [comment]
**### RCF-33** Created a credentials page for deisplaying the machine credentials on long pressing of the logo.

File: lib/demographic_details_view.dart [hunk 1]
```diff
@@ -0,0 +1,902 @@
...
+import 'package:registration_client/data/models/screen.dart';
+import 'package:registration_client/provider/global_provider.dart';
+
+import 'package:provider/provider.dart';
+
+final Color secondaryColor = Colors.blue;
+
+final Color primaryColor = Colors.blue;
+final specificJsonData = {
+  "fields": [
```
Comment (ase-101) [inline lib/demographic_details_view.dart:21]: let's not push the changes added for local testing

File: lib/login_page.dart [hunk 2]
```diff
@@ -0,0 +1,416 @@
...
+  const LoginPage({super.key});
+
+  @override
+  State<LoginPage> createState() => _LoginPageState();
+}
+
+class _LoginPageState extends State<LoginPage> {
+  bool isMobile = true;
+  static const platform =
+      MethodChannel("com.flutter.dev/keymanager.test-machine");
```
Comment (ase-101) [inline lib/login_page.dart:33]: why is this named as test-machine? instead can we name it as get-machine-keys

File: lib/login_page.dart [hunk 3]
```diff
@@ -0,0 +1,416 @@
...
+    String response;
+    Map<String, dynamic> mp;
+    try {
+      response = await platform
+          .invokeMethod("login", {'username': username, 'password': password});
+      mp = jsonDecode(response);
+      debugPrint('Login: $mp');
+    } on PlatformException catch (e) {
+      mp = {
+        'login_response': 'Login Failed..Try Again!',
```
Comment (ase-101) [inline lib/login_page.dart:115]: can we create a standard error response to the UI, and also we should return error code. and the message should be taken from i18n bundle

File: lib/login_page.dart [hunk 4]
```diff
@@ -0,0 +1,416 @@
...
+    String response;
+    Map<String, dynamic> mp;
+
+    try {
+      response = await platform
+          .invokeMethod("validateUsername", {'username': username});
+      mp = jsonDecode(response);
+    } on PlatformException catch (e) {
+      mp = {
+        'user_response': 'Failed..Try Again!',
```
Comment (ase-101) [inline lib/login_page.dart:135]: can we create a standard error response to the UI, and also we should return error code. and the message should be taken from i18n bundle

**Review by ase-101 — CHANGES_REQUESTED** [review]

File: android/clientmanager/src/main/java/io/mosip/registration/clientmanager/repository/UserDetailRepository.java [hunk 5]
```diff
@@ -33,8 +35,8 @@ public void saveUserDetail(JSONArray users) throws JSONException {
                     .findFirst();
 
             UserDetail userDetail = new UserDetail(userId);
-            userDetail.setIsDeleted(users.getJSONObject(i).getBoolean("isDeleted"));
-            userDetail.setIsActive(users.getJSONObject(i).getBoolean("isActive"));
+            userDetail.setIsDeleted(false);
+            userDetail.setIsActive(true);
```
[the remark this answers is not in the document]
Comment (Piyush7034) [reply android/clientmanager/src/main/java/io/mosip/registration/clientmanager/repository/UserDetailRepository.java]: All hardcoded flags used for local debugging removed

File: android/clientmanager/src/main/java/io/mosip/registration/clientmanager/repository/UserDetailRepository.java [hunk 6]
```diff
@@ -56,6 +58,7 @@ public void saveUserDetail(JSONArray users) throws JSONException {
     }
 
     public boolean isActiveUser(String userId) {
+        Log.e(getClass().getSimpleName(), "All users: " + userDetailDao.getUserDetail(userId));
```
[the remark this answers is not in the document]
Comment (Piyush7034) [reply android/clientmanager/src/main/java/io/mosip/registration/clientmanager/repository/UserDetailRepository.java]: All logs used for local debug removed

File: android/clientmanager/src/main/java/io/mosip/registration/clientmanager/service/MasterDataServiceImpl.java [hunk 7]
```diff
@@ -436,7 +436,9 @@ public void onFailure(Call<ResponseWrapper<UserDetailResponse>> call, Throwable
 
     private void saveUserDetails(String encData) {
         try {
+            Log.e(TAG, "Enc Data: " + encData);
             userDetailRepository.saveUserDetail(getDecryptedDataList(encData));
+            Log.e(TAG, "User Details Count " + userDetailRepository.isActiveUser("piyush"));
```
[the remark this answers is not in the document]
Comment (Piyush7034) [reply android/clientmanager/src/main/java/io/mosip/registration/clientmanager/service/MasterDataServiceImpl.java]: All loge for local debug removed

File: android/scanlibrary/build.gradle [hunk 8]
```diff
@@ -24,6 +24,15 @@ android {
         sourceCompatibility JavaVersion.VERSION_1_8
         targetCompatibility JavaVersion.VERSION_1_8
     }
+
```
[the remark this answers is not in the document]
Comment (Piyush7034) [reply android/scanlibrary/build.gradle]: scanlibrary package removed. May be added later when it is needed

File: assets/data.json [hunk 9]
```diff
@@ -0,0 +1,1202 @@
+{
```
[the remark this answers is not in the document]
Comment (Piyush7034) [reply assets/data.json]: UI spec is removed as the files using it for testing are also removed.

File: assets/specific_data.json [hunk 10]
```diff
@@ -0,0 +1,767 @@
+{
```
[the remark this answers is not in the document]
Comment (Piyush7034) [reply assets/specific_data.json]: UI spec specific_data.json also removed

File: lib/demographic_details_view.dart (hunk 1 above)
Comment (Piyush7034) [reply lib/demographic_details_view.dart]: Removed local testing code

File: lib/login_page.dart (hunk 2 above)
Comment (Piyush7034) [reply lib/login_page.dart]: Renamed it as "io.mosip.get-package-instance"  as only single method channel is being used.

File: lib/login_page.dart (hunk 3 above)
Comment (Piyush7034) [reply lib/login_page.dart]: Created a login response model. Where error code is hardcoded for now and added messages in the l10n files.

File: lib/login_page.dart (hunk 4 above)
Comment (Piyush7034) [reply lib/login_page.dart]: Created a login response model. Where error code is hardcoded for now and added messages in the l10n files.

File: android/app/src/main/java/io/mosip/registration_client/AboutActivityService.java [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+        if(details.get("name") == null) {
+            result.error("ERR-INITIALIZATION-FAILED",
+                    "Machine details could not be fetched!",
+                    "Failed to initialize the machine. Try to clear the app data and cache before reinstalling the app.");
+        }
+
+        JSONObject jsonObject = new JSONObject(details);
+
+        try {
+            jsonObject.put("version", "Alpha");
```
Comment (ase-101) [inline android/app/src/main/java/io/mosip/registration_client/AboutActivityService.java:34]: can we take the version from the manifest file?

File: android/app/src/main/java/io/mosip/registration_client/MainActivity.java [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+import io.mosip.registration.clientmanager.spi.JobTransactionService;
+import io.mosip.registration.clientmanager.spi.MasterDataService;
+import io.mosip.registration.clientmanager.spi.PacketService;
+import io.mosip.registration.clientmanager.spi.RegistrationService;
+import io.mosip.registration.clientmanager.spi.SyncRestService;
+import io.mosip.registration.clientmanager.util.SyncRestUtil;
+import io.mosip.registration.keymanager.spi.ClientCryptoManagerService;
+
+public class MainActivity extends FlutterActivity {
+    private static final String CHANNEL_TEST = "com.flutter.dev/io.mosip.get-package-instance";
```
Comment (ase-101) [inline android/app/src/main/java/io/mosip/registration_client/MainActivity.java:35]: let us change the variable name

File: android/settings.gradle [hunk 13]
```diff
@@ -0,0 +1,17 @@
+include ':app'
+include ':clientmanager'
+include ':packetmanager'
+include ':keymanager'
+project(':clientmanager').projectDir = new File('clientmanager')
+project(':packetmanager').projectDir = new File('packetmanager')
+project(':keymanager').projectDir = new File('keymanager')
+
+def localPropertiesFile = new File(rootProject.projectDir, "local.properties")
```
Comment (ase-101) [inline android/settings.gradle:9]: where do we have the local.properties file?

**Review by ase-101 — APPROVED** [review]

**Pull request merged**